website/blog/detox-20-is-out.md

```diff
@@ -0,0 +1,260 @@
+module.exports = {
+  // ...
+  apps: {
+    'android.debug': {
+      type: 'android.apk',
+      binaryPath: '...',
+      reversePorts: [3000],
+    },
+  },
+};
+```
+
+In other words, this is a convenience API that tells Detox to run `device.reverseTcpPort(portNumber)` after installing the app. It should be helpful for anyone who prefers to keep such things as configs rather than as code.
+
+### Re-launching apps
+
+As we can tell from production experience, using `device.launchApp()` as-is, without passing `newInstance` parameter explicitly, is a source of flakiness in some tests.
+
+The default `newInstance` value is conditional, in fact – if the app is already running, this method just brings it to the foreground, otherwise, it launches a new instance.
+That often creates undesirable situations, e.g., when a test might restart the app unexpectedly, just because the app had crashed in the previous test.
+
+We plan to change the default behavior in the next major release, so that the `launchApp` will always be re-launching the app, unless you pass `newInstance: false` explicitly, when you want to bring it to the foreground. Therefore, we suggest you to switch everywhere to the new method, `device.relaunchApp()`, which is equivalent to `launchApp({ newInstance: true })`, except for cases when you want to bring the app to the foreground:
+
+```js
+await device.relaunchApp();
+await device.relaunchApp({ ...opts });
+await device.launchApp({ newInstance: false }); // make it explicit
+```
+
+This way, you will prepare to the upcoming breaking change in Detox 21.0.0, and you will also get rid of the inconsistency in your tests. All that would be left is to mass-replace `deivce.relaunchApp()` with `device.launchApp()` in your codebase.
+
+### Read-only emulators by default
+
+
+The `-read-only` flag appeared in [Android emulator 28.0.16](https://developer.android.com/studio/releases/emulator#concurrent-avd). Detox promptly adopted it since the read-only mode allowed it to run multiple instances of a single Android virtual device (AVD) concurrently. This feature helped us to implement parallel test execution support for Android back then.
+
+Being overcautious, we implemented that partially, only for cases when the user starts multiple concurrent workers. This decision created a moderately annoying UX issue. Imagine you run tests sequentially first, using one worker only. That provides you with a regular AVD instance, i.e., not a read-only one. After that, you switch to multiple workers only to get an error from the Android emulator, complaining about mixing regular and read-only instances.
+
+While the fix itself has always been straightforward – close the running AVD and try again – this entire overcaution brought more issues than solving them. That's why, from now on, Android emulators will always be starting in `-read-only` mode unless you configure `readonly: false` in your [device config][`Config file > Device`].
+
+### Reset lock file
```

Comment:
Might be a good idea to mentioned that we have plans for getting rid of lock files in future versions, etc., hence this is a manageable solution for a temporary inconvenience rather than a desired long-term feature.